llvm/lib/Target/EVM/EVMAsmPrinter.cpp

```diff
@@ -70,12 +76,43 @@ void EVMAsmPrinter::emitFunctionEntryLabel() {
 void EVMAsmPrinter::emitInstruction(const MachineInstr *MI) {
   EVMMCInstLower MCInstLowering(OutContext, *this, VRegMapping,
                                 MF->getRegInfo());
-
   MCInst TmpInst;
   MCInstLowering.Lower(MI, TmpInst);
   EmitToStreamer(*OutStreamer, TmpInst);
 }
 
+bool EVMAsmPrinter::isBlockOnlyReachableByFallthrough(
+    const MachineBasicBlock *MBB) const {
+  if (!AsmPrinter::isBlockOnlyReachableByFallthrough(MBB))
+    return false;
+
+  // Check terminators in more details. After the stackification we may have the
+  // following case:
+
+  //   PUSH_LABEL %bb.11
+  //   JUMPI %bb.11, %27:gpr..
+  //   PUSH_LABEL %bb.9
+  //   JUMP %bb.9
+  // bb.11:
+  //
+  // The default implementation checks only the first terminator, JUMP %bb.9
```

Comment:
> the first terminator

the last one if to read instructions as they follow.

Reply:
I changed the description.